src/components/product-switcher/index.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+  KeyboardEventHandler,
+  ReactElement,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+import { useRouter } from 'next/router'
+import classNames from 'classnames'
+import { IconCaret16 } from '@hashicorp/flight-icons/svg-react/caret-16'
+import { Product } from 'common/types'
+import ProductIcon from 'components/icons/product-icon'
+import s from './style.module.css'
+
+const OPTION_LIST_ID = 'product-chooser-option-list'
+const OPTION_ID_PREFIX = 'product-chooser-list-item-'
+
+// TODO: should we put this somewhere else for easier reuse?
+const products: Product[] = [
+  {
+    name: 'Terraform',
+    code: 'terraform',
+    url: 'https://www.terraform.io/docs/index.html',
+  },
+  {
+    name: 'Packer',
+    code: 'packer',
+    url: 'https://www.packer.io/docs',
+  },
+  {
+    name: 'Consul',
+    code: 'consul',
+    url: 'https://www.consul.io/docs',
+  },
+  {
+    name: 'Vault',
+    code: 'vault',
+    url: 'https://www.vaultproject.io/docs',
+  },
+  {
+    name: 'Boundary',
+    code: 'boundary',
+    url: 'https://www.boundaryproject.io/docs',
+  },
+  {
+    name: 'Nomad',
+    code: 'nomad',
+    url: 'https://www.nomadproject.io/docs',
+  },
+  {
+    name: 'Waypoint',
+    code: 'waypoint',
+    url: '/waypoint/docs',
+  },
+  {
+    name: 'Vagrant',
+    code: 'vagrant',
+    url: 'https://www.vagrantup.com/docs',
+  },
+  {
+    name: 'Sentinel',
+    code: 'sentinel',
+    url: 'https://docs.hashicorp.com/sentinel',
+  },
+  {
+    name: 'HashiCorp Cloud Platform',
+    code: 'hcp',
+    url: 'https://cloud.hashicorp.com/docs/hcp',
+  },
+]
+
+const generateSwitcherOptionIdFromProduct = (product: Product) => {
+  return `${OPTION_ID_PREFIX}${product.code}`
+}
+
+const getAllAnchorElements = () => {
+  const listElement = document.getElementById(OPTION_LIST_ID)
+  const anchorElements = listElement.querySelectorAll('a')
+  return anchorElements
+}
+
+const ProductSwitcher: React.FC = () => {
```

Comment:
Just thinking out loud here...there's a lot of (good) code here to ensure we've got all of our accessibility requirements handled, did we look assess any of the Reach components to see if we might be able to get some of that out of the box? This one looks like it could potentially be valuable: https://reach.tech/menu-button

I see your comment about not using `role="menu"`, so maybe that's why we're doing this custom 💭

Reply:
Yep! That's exactly why I didn't use that component. 👍🏻